test/ecto/adapters/postgres_test.exs

```diff
@@ -669,6 +669,16 @@ defmodule Ecto.Adapters.PostgresTest do
            ~s|CREATE UNIQUE INDEX "posts_permalink_index" ON "posts" ("permalink")|
   end
 
+  test "create unique index with condition" do
+    create = {:create, index(:posts, [:permalink], unique: true, where: "public IS TRUE")}
+    assert SQL.execute_ddl(create) ==
+           ~s|CREATE UNIQUE INDEX "posts_permalink_index" ON "posts" ("permalink") WHERE public IS TRUE|
```

Comment:
note, I would normally do one case per test
but this style was copied from the `test "create index with prefix"` above